File: lib/domain/repositories/search_history_repository.dart
import 'package:libcheck/domain/models/search_history_entry.dart';

abstract class SearchHistoryRepository {
  Future<List<SearchHistoryEntry>> getAll();
  Future<void> save(SearchHistoryEntry entry);
  Future<void> remove(String isbn);
  Future<void> removeAll();
}

File: src/domain/repositories/searchHistoryRepository.ts
import { SearchHistoryEntry } from '@/domain/models/searchHistoryEntry';

export interface SearchHistoryRepository {
  getAll(): Promise<SearchHistoryEntry[]>;
  save(entry: SearchHistoryEntry): Promise<void>;
  remove(isbn: string): Promise<void>;
  removeAll(): Promise<void>;
}
